coordinator/persistence-api/src/main/java/io/stargate/db/schema/VectorType.java

```diff
@@ -0,0 +1,32 @@
+package io.stargate.db.schema;
+
+/**
+ * Actual type definition for {@code Column.Type.Vector}: needed because we need to pass
+ * "dimensions" parameter, and cannot quite use {@link ParameterizedType} since parameterization is
+ * not by type (yet at least?) but by fixed length.
+ */
+public class VectorType extends Column.DelegatingColumnType {
+  private final int dimensions;
+
+  private final String marshalTypeName;
+
+  protected VectorType(int dimensions) {
+    super(Column.Type.Vector);
+    this.dimensions = dimensions;
+    marshalTypeName = String.format("org.apache.cassandra.db.marshal.VectorType(%s)", dimensions);
```

Comment:
```suggestion
    marshalTypeName = String.format("org.apache.cassandra.db.marshal.VectorType(org.apache.cassandra.db.marshal.FloatType, %s)", dimensions);
```
This is the new syntax post CASSANDRA-18504

Reply:
Committed a change but now thinking it might not work as expected; probably need method different. But would like to have it pluggable regardless.